### PR TITLE
Restructure alloc space calls into simpler smaller files

### PR DESCRIPTION
**Pull request by islas**
TYPE: enhancement

KEYWORDS: intel, compilation, llvm, memory

SOURCE: internal

DESCRIPTION OF CHANGES:
Problem:
The Intel oneAPI compilers (and others like nvhpc) struggle with some of the larger (15k+ lines of code) files within WRF. This causes intense memory usage that is not often available to the average user not in a resource-rich environment. This often limits compilation to single threaded if even possible or to a dedicated environment with enough memory if available. If neither of those is available to a user, they will be unable to use these configurations entirely.

Solution:
This PR focuses on the `module_alloc_space*` sections of code to reduce their individual file size to manageable levels. They are instead broken out into many smaller files as external subroutines, not requiring a wrapper module to house the subroutine call. The files are now fully generated source code from the registry, with the calls to the subroutines also being generated as well. This also makes it relatively easy to change the number of files generated from a source code perspective. Build rules would need to be modified accordingly as seen in these changes.

TESTS CONDUCTED: 
Attached to this PR are plots of the respective effects of theses changes. Changes were tested with intel and gcc compilers, but only intel memory usage is shown as it exacerbates the memory usage issue.

### CODE REVIEW

**Comment by islas** [comment]
Highlighted is the region during compilation which memory usage spikes that this PR addresses (`module_alloc_space*`) before these changes take place :
pre_alloc

This usage is then dropped when using this PR's edits :
post_alloc_split

Zooming in we can now see that allocs_ comprise a number of smaller compilation units :
post_alloc_split_zoom

**Comment by weiwangncar** [comment]
@islas This PR has not passed Jenkins tests at this time. The failed test is with the DA compilation.

**Comment by weiwangncar** [comment]
I tested code before and after this PR, and model produces identical results in my test. Also with 4 processors, the compile time is about 12 minutes!

**Comment by weiwangncar** [comment]
The regression test results:

    Test Type              | Expected  | Received |  Failed
    = = = = = = = = = = = = = = = = = = = = = = = =  = = = =
    Number of Tests        : 23           24
    Number of Builds       : 60           57
    Number of Simulations  : 158           150        0
    Number of Comparisons  : 95           86        0

    Failed Simulations are: 
    None
    Which comparisons are not bit-for-bit: 
    None

**Review by weiwangncar — APPROVED** [review]

**Review by mgduda — CHANGES_REQUESTED** [review]
Just a few minor suggestions. Those aside, this looks good to me.

File: var/build/makefile [hunk 1]
```diff
@@ -71,4 +71,3 @@ libwrfvar.a : $(WRFVAR_OBJS)
           if [ -n "$(OMP)" ] ; then echo COMPILING $*.f90 WITHOUT OMP ; fi ; \
 	  $(FC) -o $@ -c $(FCFLAGS) $(PROMOTION) $*.f ; \
         fi
-
```
[the remark this answers is not in the document]
Comment (islas) [reply var/build/makefile:74]: Ah, must have deleted an extra line when getting rid of the .F90 rule I added then removed.

**Review by mgduda — APPROVED** [review]

**Review by weiwangncar — APPROVED** [review]

**Pull request merged**